src/parser/parser.h

```diff
@@ -105,132 +105,158 @@ inline std::ostream& operator<<(std::ostream& out, SymbolType type) {
 class CVC5_EXPORT Parser
 {
   friend class ParserBuilder;
-private:
 
- /** The input that we're parsing. */
- std::unique_ptr<Input> d_input;
+ private:
+  /** The input that we're parsing. */
+  Input* d_input;
 
- /**
-  * Reference to the symbol manager, which manages the symbol table used by
-  * this parser.
-  */
- SymbolManager* d_symman;
+  /**
+   * Reference to the symbol manager, which manages the symbol table used by
+   * this parser.
+   */
+  SymbolManager* d_symman;
 
- /**
-  * This current symbol table used by this parser, from symbol manager.
-  */
- SymbolTable* d_symtab;
+  /** The language that we are parsing. */
+  InputLanguage d_lang;
```

Comment:
Why do we suddenly need this information here? (i.e.: where did it live beforehand?)